packages/cli/src/controllers/mfa.controller.ts

```diff
@@ -64,18 +69,18 @@ export class MFAController {
 		}
 
 		const verified = this.mfaService.totp.verifySecret({ secret, token, window: 10 });
-
 		if (!verified)
 			throw new BadRequestError('MFA token expired. Close the modal and enable MFA again', 997);
 
-		await this.mfaService.enableMfa(id);
+		const user = await this.mfaService.enableMfa(id);
```

Comment:
Here, we are returning the saved user without the secret, probably because the secret is not selectable. Because of this, the hash is created with an empty secret. Hence, you get logged out if you refresh right after enabling MFA because the hash we have in the JWT does not match the hash we create within `resolveJwt`. I guess here, we need to query the user's secret again and add it to the user's object before calling `issueCookie`.

https://www.loom.com/share/141057d565a343578c30051779b1cbcb

Reply:
Yeah. the `select: false` on columns, and `.find` calls with a `select` param create a bunch of issues like this because we have no type-guarantees on the returned objects.
I've been trying to separate out the auth related properties out of the `User` entity, and I'm currently on my 4th attempt to do so. Turns out fixing this retroactively is quite messy, and I wish we had broken the `User` entity much earlier.

Reply:
finally managed to finish that PR https://github.com/n8n-io/n8n/pull/9557.
Once that PR is merged, I'll update this PR.